js/modules/loki_app_dot_net_api.js

```diff
@@ -418,6 +576,55 @@ class LokiAppDotNetServerAPI {
     this.channels.splice(i, 1);
   }
 
+  // set up pubKey & pubKeyHex properties
+  // optionally called for mainly file server comms
+  getPubKeyForUrl() {
+    // Hard coded
+    let pubKeyAB;
+    if (urlPubkeyMap) {
+      pubKeyAB = window.Signal.Crypto.base64ToArrayBuffer(
+        urlPubkeyMap[this.baseServerUrl]
+      );
+    }
+    // else will fail validation later
+
+    // do we have their pubkey locally?
+    // FIXME: this._server won't be set yet...
+    // can't really do this for the file server because we'll need the key
+    // before we can communicate with lsrpc
+    /*
+    // get remote pubKey
+    this._server.serverRequest('loki/v1/public_key').then(keyRes => {
+      // we don't need to delay to protect identity because the token request
+      // should only be done over lokinet-lite
+      this.delayToken = true;
+      if (keyRes.err || !keyRes.response || !keyRes.response.data) {
+        if (keyRes.err) {
+          log.error(`Error ${keyRes.err}`);
+        }
+      } else {
+        // store it
+        this.pubKey = dcodeIO.ByteBuffer.wrap(
+          keyRes.response.data,
+          'base64'
+        ).toArrayBuffer();
+        // write it to a file
+      }
+    });
+    */
+
+    // now that key is loaded, lets verify
+    if (pubKeyAB && pubKeyAB.byteLength && pubKeyAB.byteLength !== 33) {
+      log.error('FILESERVER PUBKEY is invalid, length:', pubKeyAB.byteLength);
+      process.exit(1);
```

Comment:
Uh should process really exit?? can't we throw an error or something?

Reply:
Session won't function properly if someone messed with their configuration in this way.

Reply:
How would they know how to recover from a weird configuration?

Reply:
They would have to reach out and we'd diagnosis it as such. I mean an alert or something would be better than what we have, but not sure how we that with our UI.

Reply:
> They would have to reach out and we'd diagnosis it as such

i don't think this is good UX at all. 90% of the time, the non-technical people won't reach out and they will just stop using session

Reply:
Should it match what happens when a JSON config is corrupt?
![Screen Shot 2020-05-24 at 3 58 19 PM](https://user-images.githubusercontent.com/1224671/82766857-6ff3ce80-9dd7-11ea-9295-cc0166cb84b5.png)

Reply:
Oh so `pubkeyAB` is a key that we set in the config? I thought it was user set. In that case ye this should be fine.

Reply:
Yes, it's a hardcoded value (or lack of one) based on the file server URL in your config. So the only time we wouldn't have a hardcoded value is if someone put a file server URL without updating the hardcode table.